### ISSUE TITLE
Some bugs or improvements to fix in the music (and SFX) library

### ISSUE THREAD

**Issue by Fenix668** [MegaSa1nt/GMDprivateServer#221]
### Improvement
- add for the NCS library all the other authors who collaborated with some songs, and why not, allow reupload songs to add more than one author or the same tags (specifying a limited number to put them) that are in the library.

### Bugs
- fix the issue with the SFX library in the main "Geometry Dash" folder, which currently only shows 4 folders instead of the expected 22 in the main folder.
- fix the duration value that the songs (and even if you can on SFX where the milliseconds value does not come out) in the library where it is not displayed (some even have "0s" left).
- fix for version 2.206 the names of authors of reupload songs no longer appearing in the authors list (the list below the filter list).

**Comment by MegaSa1nt** [comment]
1. Several authors are already in the core
2. I'll check
3. Check if your server can run getID3.php
4. Can you show screenshot?

**Comment by Fenix668** [comment]
With the first point I mean the authors who have collaborated for other songs that come out with "+1". It's a new update made in version 2.206, because before the authors were single.
On my server I have all the getID3.php files, so I don't know why it returns 0 in the "duration" value. As for the last point, here is the screen, I tried to add 2/3 and this is the result:

<img width="601" alt="Screen 2024-09-07 110009" src="https://github.com/user-attachments/assets/a887ad1c-3e45-45c1-94a5-743c38c8c818">
<img width="701" alt="Screen 2024-09-07 105218" src="https://github.com/user-attachments/assets/8c3d6b4d-c883-49ae-88e3-4edc3b240107">

The only thing that is readable is the GDPS folder of the reuploaded songs and the name of the players who uploaded the songs on the filter list. While the other list comes out empty only for songs reuploaded from own server.

**Comment by MegaSa1nt** [comment]
1. It is already in the core, but not for reuploaded songs

**Comment by Fenix668** [comment]
Then I check again if there have been any updates on mainLib or incl/misc/getGJSongInfo.php to see for 1.

**Comment by MegaSa1nt** [comment]
I forgot to push commit to first LOL, its in my gdps currently

**Comment by Fenix668** [comment]
Ah that's why I couldn't find it XD

**Comment by Fenix668** [comment]
But don't worry, you can commit later as soon as you manage to solve the other points as soon as you check.

**Comment by MegaSa1nt** [comment]
another suggestion: add icon to `Choose chat` text, same as `No comments` text 
![image](https://github.com/user-attachments/assets/340938b5-541d-48c6-9b2c-fbcf669c9743)

**Comment by Fenix668** [comment]
Regarding point 2, I was wondering if you could also add the duration on the dashboard to the custom songs. Like 1 m 43 s. While on the level list part add NCS mark (similar to what you see on GD, with the small NCS icon next to the song) also on the songs, and in order not to include all the authors, if there are more, leave one with the number of other authors who collaborated (such as +1 or +2).

**Comment by Fenix668** [comment]
I always leave it below but I don't know if it's a bug or a problem with Newgrounds, but in the dashboard the audio button doesn't play the songs coming from that platform (Newgrounds).

**Comment by MegaSa1nt** [comment]
> I always leave it below but I don't know if it's a bug or a problem with Newgrounds, but in the dashboard the audio button doesn't play the songs coming from that platform (Newgrounds).

Newgrounds doesn't allow listening songs from their website on other sites

**Comment by MegaSa1nt** [comment]
Added several authors to music library
New songs should have duration now

**Comment by Fenix668** [comment]
Thanks for the commit. I'll try to see

**Comment by Fenix668** [comment]
Checked. Now the duration value is returned, each new song comes out with the writing "new" as for the NCS library. But there is a small bug that concerns the songs with single authors, now for every song, even where there are single artists, "+1" comes out even when there is only one artist:


https://github.com/user-attachments/assets/9702d57a-4605-4a05-bb3f-db1c0034aa49

**Comment by MegaSa1nt** [comment]
lol

**Comment by MegaSa1nt** [comment]
fixed

**Comment by Fenix668** [comment]
I have removed the improvements that have been made from the issue. One thing, but should the several authors also be read when you open the levels, as well as in the library?

**Comment by MegaSa1nt** [comment]
yes, but i didn't implement it yet

**Comment by Fenix668** [comment]
Ah ok, for now we only read one author, I'll leave this problem anyway as soon as you slowly implement everything

**Comment by Fenix668** [comment]
As for the SFX folder bug, I noticed that the issue came after the changes in this commit. Try to restore by putting the changes back as they were before and tell me if the folders return 22 instead of 4: https://github.com/MegaSa1nt/GMDprivateServer/commit/7dd36d7ccb75dc47db08baf78d9fade33518613b

**Comment by Fenix668** [comment]
(sorry, I rewrote it here because I got the wrong issue lol)

**Comment by MegaSa1nt** [comment]
> As for the SFX folder bug, I noticed that the issue came after the changes in this commit. Try to restore by putting the changes back as they were before and tell me if the folders return 22 instead of 4: https://github.com/MegaSa1nt/GMDprivateServer/commit/7dd36d7ccb75dc47db08baf78d9fade33518613b

Yeah, situation is semi-bad, at one side i have 4 folders, at other, dublicating sfxs

**Comment by MegaSa1nt** [comment]
> As for the SFX folder bug, I noticed that the issue came after the changes in this commit. Try to restore by putting the changes back as they were before and tell me if the folders return 22 instead of 4: [7dd36d7](https://github.com/MegaSa1nt/GMDprivateServer/commit/7dd36d7ccb75dc47db08baf78d9fade33518613b)

can you check this now?

**Comment by Fenix668** [comment]
> > As for the SFX folder bug, I noticed that the issue came after the changes in this commit. Try to restore by putting the changes back as they were before and tell me if the folders return 22 instead of 4: [7dd36d7](https://github.com/MegaSa1nt/GMDprivateServer/commit/7dd36d7ccb75dc47db08baf78d9fade33518613b)
> 
> can you check this now?

Yeah, I saw the update and now you can see all the folders. I had already removed it from the bugs. There is only one left to solve but I still haven't understood why if you click on the music icon the names of the authors to select do not appear.

**Comment by MegaSa1nt** [comment]
> if you click on the music icon the names of the authors to select do not appear.

wdym?

**Comment by Fenix668** [comment]
> * fix for version 2.206 the names of authors of reupload songs no longer appearing in the authors list (the list below the filter list).

I mean this. In version 2.206 if you click the author list it is empty. Instead only from previous versions (like 2.205) the list of authors can be seen.

**Comment by Fenix668** [comment]
Version 2.205:

https://github.com/user-attachments/assets/53a26e17-880d-476d-895e-15758dfb115a


Version 2.206:

https://github.com/user-attachments/assets/76954181-4859-4ff6-a1ee-af8690e51054

**Comment by Fenix668** [comment]
In version 2.206 the authors section is empty. Try checking, the authors don't come out

**Comment by MegaSa1nt** [comment]
![image](https://github.com/user-attachments/assets/df1c5c70-19b5-4818-be2b-f4dbda1b4b62)
i cant replicate that

**Comment by Fenix668** [comment]
Because you have to enter your server, not Geometry Dash lol

**Comment by MegaSa1nt** [comment]
thats my server lol

**Comment by MegaSa1nt** [comment]
you mean filter songs by my server?

**Comment by Fenix668** [comment]
Yes, uncheck "Geometry Dash" and put it on your server name in the filters

**Comment by MegaSa1nt** [comment]
screenshot above

**Comment by Fenix668** [comment]
As you can see in the screen, among 548 songs, not all the authors of all the songs are visible. Only 81 songs are seen (i calculated all the songs by the number), the filter does not calculate the other 467

**Comment by MegaSa1nt** [comment]
yeah, thats true, ill investigate

**Comment by Fenix668** [comment]
Not all the authors are seen, there are many of those present that the filter did not insert. Maybe it's the same problem as the sfxs folders and there's something to remove

**Comment by MegaSa1nt** [comment]
sad to tell, but this bug is impossible to fix. there is no ways. try it yourself

**Comment by Fenix668** [comment]
There is absolutely no need to worry, you have done everything possible to improve and fix bugs in the music library. In that case, if one day I notice, like the bug you didn't fix a few months ago in the sfxs library that wasn't working, (https://github.com/MegaSa1nt/GMDprivateServer/pull/143)
I'll do a PR like that time or I'll let you know. It may be that it is directly a bug or an incomplete update of 2.206 which was present in version 2.205 (or previous to this one). Let's leave it like this for now, thanks anyway.

**Comment by MegaSa1nt** [comment]
This bug is also in 2.207
